src/Http/Http.Abstractions/src/Microsoft.AspNetCore.Http.Abstractions.csproj

```diff
@@ -37,6 +37,7 @@ Microsoft.AspNetCore.Http.HttpResponse</Description>
   <ItemGroup>
     <InternalsVisibleTo Include="Microsoft.AspNetCore.Http.Abstractions.Microbenchmarks" />
     <InternalsVisibleTo Include="Microsoft.AspNetCore.Http.Extensions" />
+    <InternalsVisibleTo Include="Microsoft.AspNetCore.Http.Results" />
```

Comment:
```suggestion
```

I didn't realize not making the new `HttpValidationProblemDetails` constructor public required using `InternalsVisibleTo`. While we do use `InternalsVisibleTo` in some places, we'd rather avoid it where possible.

I'll try to get API approval for the new constructor later this week. It shouldn't be controversial. And it's better for consistency as you've already pointed out. I just left a new [comment](https://github.com/dotnet/aspnetcore/issues/41899#issuecomment-2201120635) on the issue.